### PR TITLE
Sb 01

### PR DESCRIPTION
**Pull request by jah701**

### CODE REVIEW

File: .mvn/wrapper/MavenWrapperDownloader.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    private static final String MAVEN_WRAPPER_JAR_PATH =
+            ".mvn/wrapper/maven-wrapper.jar";
+
+    /**
+     * Name of the property which should be used to override the default download url for the wrapper.
+     */
+    private static final String PROPERTY_NAME_WRAPPER_URL = "wrapperUrl";
+
+    public static void main(String args[]) {
+        System.out.println("- Downloader started");
```
Comment (fyarkiy) [inline .mvn/wrapper/MavenWrapperDownloader.java:50]: Did you consider to use Logger?

File: .mvn/wrapper/MavenWrapperDownloader.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+                url = mavenWrapperProperties.getProperty(PROPERTY_NAME_WRAPPER_URL, url);
+            } catch (IOException e) {
+                System.out.println("- ERROR loading '" + MAVEN_WRAPPER_PROPERTIES_PATH + "'");
+            } finally {
+                try {
+                    if (mavenWrapperPropertyFileInputStream != null) {
+                        mavenWrapperPropertyFileInputStream.close();
+                    }
+                } catch (IOException e) {
+                    // Ignore ...
```
Comment (fyarkiy) [inline .mvn/wrapper/MavenWrapperDownloader.java:73]: and? Why would you need this catch and not to throw any messages?

File: .mvn/wrapper/MavenWrapperDownloader.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+            }
+        }
+        System.out.println("- Downloading to: " + outputFile.getAbsolutePath());
+        try {
+            downloadFileFromURL(url, outputFile);
+            System.out.println("Done");
+            System.exit(0);
+        } catch (Throwable e) {
+            System.out.println("- Error downloading");
+            e.printStackTrace();
```
Comment (fyarkiy) [inline .mvn/wrapper/MavenWrapperDownloader.java:93]: Bohdan doesn't like printStakeTrace... Don't ask me "why" :)

**Review by fyarkiy — APPROVED** [review]
Good work

**Comment by RomanDubovskyi** [comment]
don't forget about informative pr and commit names 
image

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+            <version>1.2.17</version>
+        </dependency>
+
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-data-jpa</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-thymeleaf</artifactId>
```
Comment (RomanDubovskyi) [inline pom.xml]: remove dependencies, that you don't need now from or

File: src/test/java/com/spring/boot/ApplicationTests.java [hunk 5]
```diff
@@ -0,0 +1,13 @@
+package com.spring.boot;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+
+@SpringBootTest
+class ApplicationTests {
```
Comment (RomanDubovskyi) [inline src/test/java/com/spring/boot/ApplicationTests.java]: why do we need this class?

File: src/test/java/com/spring/boot/util/CustomFileReaderImplTest.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
+package com.spring.boot.util;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class CustomFileReaderImplTest {
```
Comment (RomanDubovskyi) [inline src/test/java/com/spring/boot/util/CustomFileReaderImplTest.java]: what about valid case, don't see tests for it

File: src/main/resources/application.properties [hunk 7]
```diff
@@ -0,0 +1,14 @@
+spring.datasource.url=jdbc:h2:mem:testdb
+spring.datasource.driverClassName=org.h2.Driver
```
Comment (RomanDubovskyi) [inline src/main/resources/application.properties]: we don't work with database now, first task is just to read from file

File: src/main/java/com/spring/boot/util/CustomFileReaderImpl.java [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CustomFileReaderImpl implements CustomFileReader {
+    @Override
+    public List<String> readFile(String path) {
+        List<String> result = new ArrayList<>();
+        String line;
+        try {
```
Comment (RomanDubovskyi) [inline src/main/java/com/spring/boot/util/CustomFileReaderImpl.java]: close resource or use TWR

File: src/main/java/com/spring/boot/util/CustomFileReaderImpl.java [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+            BufferedReader reader = new BufferedReader(new FileReader(path));
+            while ((line = reader.readLine()) != null) {
+                result.add(line);
+            }
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException("Wrong file path", e);
```
Comment (RomanDubovskyi) [inline src/main/java/com/spring/boot/util/CustomFileReaderImpl.java]: add path to message

File: src/test/java/com/spring/boot/util/CustomFileReaderImplTest.java (hunk 6 above)
Comment (jah701) [reply src/test/java/com/spring/boot/util/CustomFileReaderImplTest.java]: done

File: src/main/java/com/spring/boot/util/CustomFileReaderImpl.java (hunk 8 above)
Comment (jah701) [reply src/main/java/com/spring/boot/util/CustomFileReaderImpl.java]: done

File: .gitignore [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+.springBeans
+.sts4-cache
+
+### IntelliJ IDEA ###
+.idea
+*.iws
+*.iml
+*.ipr
+
+### NetBeans ###
```
Comment (Sofasmile) [inline .gitignore]: ми можемо спростити гітігнор

File: src/main/java/com/spring/boot/event/StartupEvent.java [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+
+    @Override
+    public void onApplicationEvent(ApplicationReadyEvent applicationReadyEvent) {
+        log.info("Startup Event method onApplicationEvent() started . . .");
+        loadCsvFile(csvFileUrl, loadedFile);
+        log.info("Startup Event method onApplicationEvent() finished successfully");
+    }
+
+    public boolean loadCsvFile(String url, String intoPath) {
+        try (BufferedInputStream in = new BufferedInputStream(new URL(url).openStream());
```
Comment (Sofasmile) [inline src/main/java/com/spring/boot/event/StartupEvent.java]: давай перейменуємо змінну in

File: src/main/java/com/spring/boot/event/StartupEvent.java [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+                FileOutputStream fileOutputStream = new FileOutputStream(intoPath)) {
+            byte[] dataBuffer = new byte[1024];
+            int bytesRead;
+            while ((bytesRead = in.read(dataBuffer, 0, 1024)) != -1) {
```
Comment (Sofasmile) [inline src/main/java/com/spring/boot/event/StartupEvent.java]: чому зчитуємо тільки по 1024 байта?

File: src/main/java/com/spring/boot/util/CustomFileReaderImpl.java [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+    @Override
+    public List<String> readFile(String path) {
+        List<String> result = new ArrayList<>();
+        String line;
+        try {
+            BufferedReader reader = new BufferedReader(new FileReader(path));
+            while ((line = reader.readLine()) != null) {
+                result.add(line);
+            }
+            reader.close();
```
Comment (Sofasmile) [inline src/main/java/com/spring/boot/util/CustomFileReaderImpl.java]: де потрібно закривати ресурс?

File: src/test/java/com/spring/boot/util/CustomFileReaderImplTest.java [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+    private static final String WRONG_PATH = "";
+    private static final String CORRECT_PATH = "src/main/resources/file.csv";
```
Comment (Sofasmile) [inline src/test/java/com/spring/boot/util/CustomFileReaderImplTest.java]: для тестів використовується окрема папка: src/test/resources/

**Pull request closed without merging**